docs: onboarding customers and users

hifabienne opened the pull request:
This guide describes how you can onboard new customers and users by either using the builtin functionality of ZITADEL or building your own form

### Definition of Ready

- [x] I am happy with the code
- [x] Short description of the feature/issue is added in the pr description
- [x] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [ ] No debug or dead code
- [x] My code has no repetitions
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [x] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.

hifabienne commented:
hei @mffap 
I have added a first draft for the articles.
So far I have added the basics, but not yet gone through all the additional information we should include.
Would be great if you can have a look at it and give a first feedback on the structure, and guide. 
If you think it makes sense like it is, I will go through the addition questions.

I can answer them directly where it makes sense.
And for those that do not relate directly to a section I would propose to add those questions at the end, and show how to solve it

mffap commented:
> hei @mffap I have added a first draft for the articles. So far I have added the basics, but not yet gone through all the additional information we should include. Would be great if you can have a look at it and give a first feedback on the structure, and guide. If you think it makes sense like it is, I will go through the addition questions.

Looks good so far. I think that will clear up a lot of things when looking at the different register functionalities.
I will provide some more detailed feedback shortly. After a first look, I'm missing only the following: 

For B2B creating orgs is only the first step to create customers. After that they have to onboard the customer end-users. I think a register form is only one solution, and to be honest, probably the last option they need. Thus I would add to this guide:
- How to setup SSO and auto-register end-users
- How to invite users to an organization (like: "invite your team members"... send an email to create a user in that org)

hifabienne commented:
> > hei @mffap I have added a first draft for the articles. So far I have added the basics, but not yet gone through all the additional information we should include. Would be great if you can have a look at it and give a first feedback on the structure, and guide. If you think it makes sense like it is, I will go through the addition questions.
> 
> Looks good so far. I think that will clear up a lot of things when looking at the different register functionalities. I will provide some more detailed feedback shortly. After a first look, I'm missing only the following:
> 
> For B2B creating orgs is only the first step to create customers. After that they have to onboard the customer end-users. I think a register form is only one solution, and to be honest, probably the last option they need. Thus I would add to this guide:
> 
> * How to setup SSO and auto-register end-users
> * How to invite users to an organization (like: "invite your team members"... send an email to create a user in that org)

I have restructured a little bit, first I show the complete flow (How to do in the console)
As a second step I show how they can continute if they want to automate with a registration page. 
Let me know what you think

mffap commented on docs/docs/guides/manage/console/_add_manager.mdx:
```diff
@@ -0,0 +1,6 @@
+When adding a new manager, you can select multiple roles some of which are only allowed to read data.
+This can be especially useful if you add service users for one of your projects where you only need read access.
+
+Per default you will only search for users within the selected organization. If you like to give a role to a user outside the organization you need to switch to the global search and type the exact loginname of the users. This will prevent allowing users to guess users from other organizations.
```
```suggestion
Per default you will only search for users within the selected organization. If you like to give a role to a user outside the organization you need to switch to the global search and type the exact loginname of the users. This will prevent users from guessing users from other organizations.
```

mffap commented on docs/docs/guides/solution-scenarios/onboarding/_org_login_description.mdx:
```diff
@@ -0,0 +1,17 @@
+Per default the login ui of the default organization is shown.
+To allow users of a specific organization to register with their IDP (e.g Azure AD) you have to options.
```
```suggestion
To allow users of a specific organization to register with their IDP (e.g Azure AD) you have two options.
```

mffap commented on docs/docs/guides/solution-scenarios/onboarding/b2b.mdx:
```diff
@@ -0,0 +1,212 @@
+When you want to build an onboarding process for your business customers you have to go through the following steps:
+1. Create an organization: The organization represents the customer or a team
+2. Create the first administrator user: This user is the account for your customer, which should be able to configure settings such as SSO, MFA, etc.
+3. Give the user the correct permission to be able to configure settings, create users and assign roles to users in ZITADEL
```
```suggestion
3. Give the user permission to configure settings, create users and assign roles to users in ZITADEL
```

mffap commented on docs/docs/guides/solution-scenarios/onboarding/b2b.mdx:
```diff
@@ -0,0 +1,212 @@
+2. Create the first administrator user: This user is the account for your customer, which should be able to configure settings such as SSO, MFA, etc.
+3. Give the user the correct permission to be able to configure settings, create users and assign roles to users in ZITADEL
+
+As soon as you have the customer and the first administrator ready. Your customers are able to start managing.
```
```suggestion
As soon as you have successfully created the organization and added an administrator, then your customer is able to start managing the organization and users themselves.
```

mffap commented on docs/docs/guides/solution-scenarios/onboarding/b2b.mdx:
```diff
@@ -0,0 +1,212 @@
+3. Give the user the correct permission to be able to configure settings, create users and assign roles to users in ZITADEL
+
+As soon as you have the customer and the first administrator ready. Your customers are able to start managing.
+The first actions they want to take are the following:
```
```suggestion
The first actions they typically want to take are the following:
```

mffap commented on docs/docs/guides/solution-scenarios/onboarding/b2b.mdx:
```diff
@@ -0,0 +1,212 @@
+3. Configure Security Settings
+4. Configure Branding
+
+ZITADEL does have multiple possibilities on how to achieve that process.
```
```suggestion
ZITADEL does have multiple possibilities to achieve that process.
```

mffap commented on docs/docs/guides/solution-scenarios/onboarding/b2b.mdx:
```diff
@@ -0,0 +1,212 @@
+
+ZITADEL does have multiple possibilities on how to achieve that process.
+
+## Onboard in ZITADEL Management Console
```
```suggestion
## Onboard customers through the ZITADEL Management Console
```

mffap commented on docs/docs/guides/solution-scenarios/onboarding/b2b.mdx:
```diff
@@ -0,0 +1,212 @@
+
+## Onboard in ZITADEL Management Console
+
+Each ZITADEL instance does automatically bring a management console with it. This can be used to manage all your resources through a UI.
```
```suggestion
Each ZITADEL instance does automatically bring a management console with it. The console can be used to manage all your resources through a UI.
```

mffap replied:
Add link to manage > console

mffap commented on docs/docs/guides/solution-scenarios/onboarding/b2b.mdx:
```diff
@@ -0,0 +1,212 @@
+
+Each ZITADEL instance does automatically bring a management console with it. This can be used to manage all your resources through a UI.
+You can access it by calling the following URL: $CUSTOM-DOMAIN/ui/
+Make sure to authenticate with a user with permissions on an instance level such as "INSTANCE_OWNER"
```
```suggestion
Make sure that your admins have a Manager role with permissions on an instance level such as "INSTANCE_OWNER"
```

mffap commented on docs/docs/guides/solution-scenarios/onboarding/b2b.mdx:
```diff
@@ -0,0 +1,212 @@
+
+### Add First Administrator
+
+You now want to create the first user for your customer and ensure the user has enough permissions to self-manage the needed configuration.
```
```suggestion
Create the first user for your customer and ensure that the user has a Manager role with enough permissions to self-manage the needed configuration.
```

mffap commented on docs/docs/guides/solution-scenarios/onboarding/b2b.mdx:
```diff
@@ -0,0 +1,212 @@
+
+<CreateUser/>
+
+#### Add Administrator Role to User
```
```suggestion
#### Add Manager Role to User
```

mffap commented on docs/docs/guides/solution-scenarios/onboarding/b2b.mdx:
```diff
@@ -0,0 +1,212 @@
+
+#### Add Administrator Role to User
+
+Now you need to give the right administrator role to your user. In this case we want to give "ORG_OWNER".
```
```suggestion
Now you need to give the right manager role to your user. In this case we want to give "ORG_OWNER".
```

mffap commented on docs/docs/guides/solution-scenarios/onboarding/b2b.mdx:
```diff
@@ -0,0 +1,212 @@
+
+:::note
+The following actions can also be configured by an Instance Administrator for their customer.
+To show how you can use the self-management possibilities it is shown from the Customers Administrator.
```
```suggestion
To show how you can use the self-management possibilities it is shown from the Customer's Administrator.
```

mffap commented on docs/docs/guides/solution-scenarios/onboarding/b2b.mdx:
```diff
@@ -0,0 +1,212 @@
+To show how you can use the self-management possibilities it is shown from the Customers Administrator.
+:::
+
+The Administrator user got an initialization E-Mail when the user got created.
```
```suggestion
The Administrator user received an initialization email after the user was created.
```

mffap commented on docs/docs/guides/solution-scenarios/onboarding/b2b.mdx:
```diff
@@ -0,0 +1,212 @@
+
+#### Auto-register users with SSO
+
+Let's assume you have configured Google as an identity provider, and you want to allow all your employees to login with the corresponding user without having to register.
```
```suggestion
Let's assume you have configured AzureAD as an identity provider, and you want to allow all your employees to login with the corresponding user without having to register.
```

mffap commented on docs/docs/guides/solution-scenarios/onboarding/b2b.mdx:
```diff
@@ -0,0 +1,212 @@
+Let's assume you have configured Google as an identity provider, and you want to allow all your employees to login with the corresponding user without having to register.
+This does need some specific configuration on your provider.
+
+1. Go to the detail page of your configured identity provider (In this example Google)
```
```suggestion
1. Go to the detail page of your configured identity provider (In this example AzureAD)
```

mffap commented on docs/docs/guides/solution-scenarios/onboarding/b2b.mdx:
```diff
@@ -0,0 +1,212 @@
+
+## Automated onboarding for your customers
+
+If you want to start automating the process of onboard your customers the following sections give you some guidance.
```
```suggestion
If you want to start automating the process of onboarding your customers the following sections give you some guidance.
```

mffap commented on docs/docs/guides/solution-scenarios/onboarding/b2b.mdx:
```diff
@@ -0,0 +1,212 @@
+<img src="/docs/img/guides/solution-scenarios/register-organization.png" alt="Register Organization Form"/>
+
+#### Disable built-in register organization form
+If you do not want to allow users to register organizations with this form, you can disable it with the following request:
```
```suggestion

If you do not want to allow users to register organizations with this form, you can disable it with the following request:
```

mffap commented on docs/docs/guides/solution-scenarios/onboarding/end-users.mdx:
```diff
@@ -0,0 +1,92 @@
+
+#### Registration with Social Login
+
+To allow your users to register with social logins you have to configure the needed external identity providers.
```
```suggestion
To allow your users to register with social logins you have to configure the external identity providers.
```
